lib/src/participant/participant.dart

```diff
@@ -139,21 +141,25 @@ abstract class Participant extends LKChangeNotifier {
   @internal
   void addTrackPublication(TrackPublication pub) {
     pub.track?.sid = pub.sid;
-    trackPublications[pub.sid] = pub;
+    //
+    trackPublications.remove(pub);
+    trackPublications.add(pub);
   }
 
   // Must implement
   Future<void> unpublishTrack(String trackSid, {bool notify = false});
 
-  Future<void> unpublishAllTracks() async {
-    final _ = List<TrackPublication>.from(trackPublications.values);
-    for (final track in _) {
-      await unpublishTrack(track.sid);
+  Future<void> unpublishAllTracks({bool notify = false}) async {
+    final trackSids = trackPublications.map((e) => e.sid).toSet();
```

Comment:
Oh `map` returns Iterable which is still referencing the same object, calling `toSet`, `toList` creates a new object (copy).
This is to avoid **Iterating while modifying** exception.

Reply:
ah ha, I didn't realized map returned an iterator..good to know.